File: 69-sqrtx/69-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        left,right=0,x+1
        while left<right:
            mid=left+(right-left)//2
            if mid*mid <=x:
                left=mid+1
            else:
                right=mid
                
        return left-1
        

File: 69-sqrtx/69-sqrtx.cpp
class Solution {
public:
    int mySqrt(long long int x) {
      long long int low = 0,high = x+1; 
        while(low<high) {
            long long int mid = low+(high -low)/2;
            if(mid*mid>x) 
                high = mid ;
            else low = mid+1;
        }
       return low-1;
    }
};